File: server/internal/models/department.go
package models

// Fakülte
type Faculty struct {
	ID     int    `json:"id" db:"faculty_id"`
	GUID   string `json:"guid" db:"faculty_guid"`
	Name   string `json:"text" db:"faculty_name"`
	NameEn string `json:"textEn" db:"faculty_name_en"`
}

// Bölüm
type Department struct {
	ID        int    `json:"id" db:"department_id"`
	FacultyID int    `json:"ustbirimid" db:"faculty_id"`
	GUID      string `json:"guid" db:"department_guid"`
	Name      string `json:"text" db:"department_name"`
	NameEn    string `json:"textEn" db:"department_name_en"`
}

File: iuc_companion/lib/data/models/department.dart
class Department {
  final int id;
  final int facultyId;
  final String guid;
  final String name;
  final String nameEn;

  Department({
    required this.id,
    required this.facultyId,
    required this.guid,
    required this.name,
    required this.nameEn,
  });

  factory Department.fromJson(Map<String, dynamic> json) {
    return Department(
      id: json['id'] ?? 0,
      facultyId: json['ustbirimid'] ?? 0, // Go tarafında json tag: "ustbirimid"
      guid: json['guid'] ?? '',
      name: json['text'] ?? '',
      nameEn: json['textEn'] ?? '',
    );
  }
}